pennylane/devices/tests/test_gates.py

```diff
@@ -358,15 +358,25 @@ def test_supported_gates_can_be_implemented(self, device_kwargs, operation):
         device_kwargs["wires"] = 4  # maximum size of current gates
         dev = qml.device(**device_kwargs)
 
-        assert hasattr(dev, "operations")
-        if operation in dev.operations:
+        if isinstance(dev, qml.Device):
+            assert hasattr(dev, "operations")
+            if operation not in dev.operations:
+                pytest.skip("operation not supported.")
+        else:
+            if ops[operation].name == "QubitDensityMatrix":
+                prog = dev.preprocess()[0]
+                tape = qml.tape.QuantumScript([ops[operation]])
+                try:
+                    prog((tape,))
+                except qml.DeviceError:
+                    pytest.skip("operation not supported on the device")
```

Comment:
do we intend on maintaining this dual-support of both interfaces? if so, it might be good to extract this into a helper function like `supports_operation()`.

Reply:
Something I've wanted to do, but we've just haven't prioritized it yet.